octodns/provider/ns1.py

```diff
@@ -806,13 +837,15 @@ def _params_for_dynamic_A(self, record):
             us_state = set()
 
             for geo in rule.data.get('geos', []):
+
                 n = len(geo)
                 if n == 8:
                     # US state, e.g. NA-US-KY
                     us_state.add(geo[-2:])
```

Comment:
Do states require the `country` filter too? I don't believe there's a seperate state filter chain item so I'm guessing it's part of the country one, but not sure?

Reply:
Yes, part of the country rules. Will add this as part of the ordering check you commented on above.